### ISSUE TITLE
Error parsing file (red mark) when having 2 or more java classes in the same java file, inside the same package, that share same class names (e.g after cloning/copying)

### ISSUE THREAD

**Issue by istinnstudio** [apache/netbeans#4145]
### Apache NetBeans version

Apache NetBeans 13

### What happened

Error parsing file (red mark) when having 2 or more java classes in the same java file, inside the same package, that share same class names (e.g after cloning/copying).
If public class then we give the file name of the class.  If other classes (more than the main one) exist in the same java file (nested or separate non public), then we get this error without any guidance to resolve the naming conflict.
Background scanning of projects could then be triggered, and this scanning time period, will be long.

### How to reproduce

Create a public class, rename the New Class to the class name
Create a non public class inside the same file, nested and/or separated.
Copy Paste the Java file to the same package from projects list.

### Did this work correctly in an earlier version?

No

### Operating System

windows 7

### JDK

17

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by mbien** [comment]
this sounds like a very specific issue which rarely occurs. I am not sure if the time investment would be worth it to teach the parser to deal with invalid files containing duplicated class names. But contributions welcome.

**Comment by istinnstudio** [comment]
It is not critical for sure. Files are not invalid if stand alone. Only if there is an other file with same class name in the same package. It would be ok if we could get a better message to focus on what the problem is. I figured this out when opening the same project in intelij, the error message was pointing to the names, here this is just a parsing error without details.

**Comment by istinnstudio** [comment]
This error is appearing at random times, but when triggered activates background scanning of projects that takes much time to complete and sometimes an other error like this one can be found, I guess the background scanning does not scan everything at once.

It  would be nice to see a better approach on multi-class files in the same package.
The red mark appears only on file names in project/package files list, never inside the actual text of the file. Random is also which one of the involved files will be marked (between 2+ conflicting files in the same package).

To sum up, this error has random behavior witch is odd, it appears to be in conjunction with background scanning of projects that sometimes finds or misses the same conflict, and marks randomly the parsing file error to a file.

**Comment by AnderDMV** [comment]
Create a project and create some classes, delete the project and create a new one with the same name.

Later:
Toolbar -- team -- history -- revert deleted, it will show all the classes deleted from the project.

I think this is part of the problem

**Comment by BMOM** [comment]
May I ask if your current problem has been resolved?

**Comment by istinnstudio** [comment]
I have deleted/renamed all those same name classes, so I do not know if it this rare issue still exist, if other changes have been placed around source, if someone can create some classes that are being described in "How to reproduce" hold them there in a temp package and never blow up, I guess it is solved, if not you can keep the package until it shows the error and confuse the scanning mechanism. There is not an immediate effect if I remember correctly, but it did came up and stayed there after in a project reopen.

**Comment by BMOM** [comment]
I have encountered a problem similar to yours. After a long period of project scanning, strange code parsing errors may occur. And it's not always the case with every scan, it's sporadic.
